src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-hard-reset-controller.ts

```diff
@@ -0,0 +1,125 @@
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import "../../../../../components/ha-svg-icon";
+import { hardResetController } from "../../../../../data/zwave_js";
+import { haStyleDialog } from "../../../../../resources/styles";
+import { HomeAssistant } from "../../../../../types";
+import { ZWaveJSHardResetControllerDialogParams } from "./show-dialog-zwave_js-hard-reset-controller";
+import { showConfirmationDialog } from "../../../../../dialogs/generic/show-dialog-box";
+
+@customElement("dialog-zwave_js-hard-reset-controller")
+class DialogZWaveJSHardResetController extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _entryId?: string;
+
+  @state() private _done = false;
+
+  public showDialog(params: ZWaveJSHardResetControllerDialogParams): void {
+    this.entry_id = params.entryId;
+  }
+
+  public closeDialog(): void {
+    this.entry_id = undefined;
+
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+    if (this._done) {
+      this._done = false;
+      history.back();
+    }
+  }
+
+  protected render() {
+    if (!this.entry_id) {
+      return nothing;
+    }
+
+    return html`<ha-dialog
+      open
+      @closed=${this.closeDialog}
+      .heading=${createCloseHeading(
+        this.hass,
+        this.hass.localize(
+          `ui.panel.config.zwave_js.hard_reset_controller.title_${
+            this._done ? "done" : "not_done"
+          }`
+        )
+      )}
+    >
+      ${this._done
+        ? html`<p>
+            ${this.hass.localize(
+              `ui.panel.config.zwave_js.hard_reset_controller.success`
+            )}
+          </p>`
+        : html`<div class="flex-container">
+              <div>
+                <ha-svg-icon
+                  .path=${mdiDeleteForever}
+                  .class="icon"
+                ></ha-svg-icon>
+              </div>
+              <p>
+                ${this.hass.localize(
+                  `ui.panel.config.zwave_js.hard_reset_controller.dialog_body`
+                )}
+              </p>
+            </div>
+            <mwc-button
+              slot="primaryAction"
+              @click=${this._hardResetController}
+            >
+              ${this.hass.localize("ui.common.continue")}
+            </mwc-button>
+            <mwc-button slot="secondaryAction" @click=${this.closeDialog}>
+              ${this.hass.localize("ui.common.cancel")}
+            </mwc-button>`}
+    </ha-dialog>`;
+  }
+
+  private async _hardResetController(): Promise<void> {
+    if (
+      await showConfirmationDialog(this, {
+        text: this.hass.localize(
+          `ui.panel.config.zwave_js.hard_reset_controller.final_warning`
+        ),
+        dismissText: this.hass.localize("ui.common.cancel"),
+        confirmText: this.hass.localize("ui.common.continue"),
+      })
+    ) {
+      await hardResetController(this.hass, this.entry_id!);
+      this._done = true;
```

Comment:
Navigate to the devices page of the config entry here:
```ts
navigate(`/config/devices/dashboard?config_entry=${this.entry_id}`)
```

Reply:
this works, but the downside is that it takes the user to potentially a new location instead of going back to where they were, which is, in my opinion better. But I am realizing part of the problem is that the history state will include the pop ups. So if I want to do this, I think I need to store the previous page from the history and then navigate to that when done. Does that sound right?

Reply:
I implemented it with your suggestion for now, but if there's another way that we can truly go back to the previous page, I think that would be the better option

Reply:
I think the best user experience would be to navigate to the device page of the newly created device, but I'm not sure how the timing is, is the new device available immediately? Maybe `hardResetController` could return the new device id or something?

Reply:
there's a delay for sure. But yeah perhaps we could wait for the config entry to reload and a new device to be registered before calling it complete and then navigate to the new device. That would be a pretty cool user experience - can the frontend receive core events? I could wait to listen for the config entry to be reloaded and for a device to be registered and then navigate

Reply:
Yes, the frontend can handle events: `hass.connection.subscribeEvents`

Reply:
OK so made the updates above. Now the only problem is that when I navigate to the new device page, it says the device doesn't exist until I refresh the page. Not sure if there's something else I can wait for to guarantee the device is available before redirecting?

Reply:
Looking at the code the page _should_ update when new devices are added... And a refresh should not be needed...

I also see that happening in my tests...

Reply:
ok so we can remove the refresh and log a bug for this maybe?

Reply:
Do you have a quick repo for me for this?